src/BankAccountApp/Account.java

```diff
@@ -0,0 +1,94 @@
+// we want an abstract class here, because we'll never create an Account object
+// we'll be creating BankAccountApp.Checking and BankAccountApp.Savings objects (that extend Account)
+
+package BankAccountApp;
+
+
+import java.util.concurrent.ThreadLocalRandom;
+
+public abstract class Account implements IRate {
+    // list common properties for savings and checking account
+    private String name;
+    private String sSN;
+
+    double initDeposit;
+    private double balance;
+    private static int index = 10000;  // base of account number, we will increment each time we create a new account
+    double rate;
+    String accountNumber;
+
+    // Constructor to set base properties and initialize the account
+    Account(String name, String sSN, double initDeposit) {
+        this.name = name;
+        this.sSN = sSN;
+        balance = initDeposit;
+        //System.out.println("NAME: " + name + " SSN: " + sSN + " BALANCE: $" + balance);
+
+        // set account number
+        index++;
+        this.accountNumber = setAccountNumber();
+        //System.out.println("ACCOUNT NUMBER: " + this.accountNumber);
+
+
+    }
+
+    public abstract double setRate();
+
+    private String setAccountNumber() {
+        // 11-digit account number with the following properties:
+        // 1 for Savings or 2 for checking, last two digits of SSN, unique 5 digit number, and random 3 digit number
+
+        String lastTwoOfSSN = sSN.substring(sSN.length() - 2, sSN.length());
+        int uniqueID = index;
+        int randomNumber = ThreadLocalRandom.current().nextInt(1, 1000);
+        String randomNumberAsString = String.format("%03d", randomNumber);  // otherwise, drops leading zeroes, giving 1 and 2 digit nums
```

Comment:
Right. Technically, having trailing zeros at the beginning of a number is not in the scope of `int` but rather whatever is displaying it as a `String`